kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/data/blockchain/BlockchainSource.java

```diff
@@ -1,291 +1,76 @@
 package com.kin.ecosystem.data.blockchain;
 
-import android.annotation.SuppressLint;
-import android.content.Context;
 import android.support.annotation.NonNull;
-import android.text.TextUtils;
-import android.util.Log;
 import com.kin.ecosystem.Callback;
-import com.kin.ecosystem.base.ObservableData;
 import com.kin.ecosystem.base.Observer;
 import com.kin.ecosystem.data.model.Payment;
-import com.kin.ecosystem.exception.InitializeException;
-import com.kin.ecosystem.util.ExecutorsUtil.MainThreadExecutor;
 import java.math.BigDecimal;
-import java.util.concurrent.atomic.AtomicInteger;
-import kin.core.Balance;
-import kin.core.EventListener;
-import kin.core.KinAccount;
-import kin.core.KinClient;
-import kin.core.ListenerRegistration;
-import kin.core.PaymentInfo;
 
-import kin.core.ResultCallback;
-import kin.core.TransactionId;
+public interface BlockchainSource {
 
-import kin.core.exception.AccountNotFoundException;
-import kin.core.exception.CreateAccountException;
-
-public class BlockchainSource implements IBlockchainSource {
-
-    private static final String TAG = BlockchainSource.class.getSimpleName();
-
-    private static volatile BlockchainSource instance;
-
-    private KinClient kinClient;
-    private KinAccount account;
-    private ObservableData<Integer> balance = ObservableData.create(0);
     /**
-     * Listen for {@code completedPayment} in order to be notify about completed transaction sent to
-     * the blockchain, it could failed or succeed.
+     *
+     * @param appID - appID - will be included in the memo for each transaction.
```

Comment:
to better see diff it would be better to rename BlockchainSource.java to BlockchainSourceImpl.java commit and then add BlockchainSource.java interface

Reply:
👍